OrleansDashboard/DashboardTelemetryConsumer.cs

```diff
@@ -87,40 +88,32 @@ public void Flush()
         {
             var grain = grainFactory.GetGrain<ISiloGrain>(localSiloDetails.SiloAddress.ToParsableString());
 
-            var counters = new List<StatCounter>();
-
-            foreach (var metric in metrics.ToArray())
-            {
-                var v = metric.Value.Current;
-                var d = metric.Value.Current - metric.Value.Last;
-
-                counters.Add(new StatCounter { Name = metric.Key, Value = v.ToString(CultureInfo.InvariantCulture), Delta = d.ToString(CultureInfo.InvariantCulture) });
-            }
-
-            foreach (var metric in timespanMetrics.ToArray())
-            {
-                var v = metric.Value.Current;
-                var d = metric.Value.Current - metric.Value.Last;
-
-                counters.Add(new StatCounter { Name = metric.Key, Value = v.ToString("c", CultureInfo.InvariantCulture), Delta = d.ToString("c", CultureInfo.InvariantCulture) });
-            }
-
-            if (counters.Count > 0)
+            var countersArray = metrics.Select(metric => new StatCounter
```

Comment:
Actually there is no performance improvement here. In Addition: You have a view more allocations.

Reply:
You could just change the signature of the grain from array to list

Reply:
"there is no performance improvement here" 
Well, if you think that two .ToArray() don't affect performance....

"You have a view more allocations"
And where ?

Reply:
We have a benchmarking project in the solution, which is great for testing perf assumptions, and evidencing improvements (we've had performance problems reported in the past).